src/provisioning-ofono.c

```diff
@@ -157,8 +153,21 @@ static void deactivate_internet_context_reply(DBusPendingCall *call,
 {
 	DBusMessage *reply = dbus_pending_call_steal_reply(call);
 	DBusError err;
-	struct modem_data *modem;
+	struct modem_data *modem = NULL;
 	struct provisioning_data *prov_data;
+	GList *keys = g_hash_table_get_keys(modem_list);
+
+	LOG("In deactivate_internet_context_reply ");
+
+	while (keys != NULL) {
+		modem = g_hash_table_lookup(modem_list, keys->data);
+		
+		if (modem->gprs_attached) {
+			LOG("Valid path found: %s",modem->path);
+			break;
+		}
+		keys = keys->next;
+	}
 
```

Comment:
What is the above code trying to do? Why is `gprs_attached` so important here, and what does it have to do with the modem for which `AddContext` call was made?